File: CS/DragDropColumns/MyRemoveColumnDropTarget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Grid.Native;
using DevExpress.Xpf.Grid;

namespace DragDropColumns
{
    class MyRemoveColumnDropTarget : RemoveColumnDropTarget
    {
        public override void Drop(System.Windows.UIElement source, System.Windows.Point pt)
        {
            GridColumnHeader columnHeader = source as GridColumnHeader;
            GridColumn column = columnHeader.DataContext as GridColumn;
            TableView view = column.View as TableView;
            if (!view.Grid.Columns.Contains(column))
                return;
            base.Drop(source, pt);
        }
    }
}

File: VB/DragDropColumns/MyRemoveColumnDropTarget.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpf.Grid.Native
Imports DevExpress.Xpf.Grid

Namespace DragDropColumns
	Friend Class MyRemoveColumnDropTarget
		Inherits RemoveColumnDropTarget
		Public Overrides Sub Drop(ByVal source As System.Windows.UIElement, ByVal pt As System.Windows.Point)
			Dim columnHeader As GridColumnHeader = TryCast(source, GridColumnHeader)
			Dim column As GridColumn = TryCast(columnHeader.DataContext, GridColumn)
			Dim view As TableView = TryCast(column.View, TableView)
			If (Not view.Grid.Columns.Contains(column)) Then
				Return
			End If
			MyBase.Drop(source, pt)
		End Sub
	End Class
End Namespace
